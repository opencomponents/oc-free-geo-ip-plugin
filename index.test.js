const fetch = require('node-fetch');
const index = require('./');
const payload = require('./payload');

jest.setMock('node-fetch', fetch);

describe('index', () => {
  describe('register', () => {
    test('next to have been called', () => {
      const next = jest.fn();
      index.register(null, null, next);
      expect(next).toHaveBeenCalled();
    });
  });

  describe('execute', () => {
    test('fetch response to match snapshot', (done) => {
      fetch.mockClear();
      fetch.mockResponse(JSON.stringify(payload));
      const context = { requestHeaders: { 'x-forwarded-for': '98.6.106.250' } };
      index.execute(context).then((data) => {
        expect(data).toMatchSnapshot();
        done();
      });
    });

    const scenarios = [
      {
        description: 'w/ "x-forwarded-for"',
        requestHeaders: { 'x-forwarded-for': '98.6.106.250' }
      },
      {
        description: 'w/ "remoteAddress"',
        requestHeaders: { remoteAddress: '8.8.8.8' }
      },
      {
        description: 'w/ "x-forwarded-for" w/ multiple addresses',
        requestHeaders: { 'x-forwarded-for': '127.0.0.1, 192.168.0.1, 8.8.8.8' }
      },
      {
        description: 'w/ "remoteAddress" w/ multiple addresses',
        requestHeaders: { remoteAddress: '10.0.0.10, 192.168.0.1, 8.8.8.8' }
      }
    ];

    scenarios.forEach(({ description, requestHeaders }) => {
      test(description, (done) => {
        fetch.mockClear();
        fetch.mockResponse(JSON.stringify(payload));
        const context = { requestHeaders };
        index.execute(context).then(() => {
          expect(fetch.mock.calls).toMatchSnapshot();
          done();
        });
      });
    });
  });
});
