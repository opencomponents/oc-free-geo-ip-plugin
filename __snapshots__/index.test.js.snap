// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index execute fetch response to match snapshot 1`] = `
Object {
  "city": "Cedar Park",
  "country_code": "US",
  "country_name": "United States",
  "ip": "98.6.106.250",
  "latitude": 30.4998,
  "longitude": -97.8082,
  "metro_code": 635,
  "region_code": "TX",
  "region_name": "Texas",
  "time_zone": "America/Chicago",
  "zip_code": "78613",
}
`;

exports[`index execute w/ "remoteAddress" 1`] = `
Array [
  Array [
    "http://freegeoip.net/json/8.8.8.8",
  ],
]
`;

exports[`index execute w/ "remoteAddress" w/ multiple addresses 1`] = `
Array [
  Array [
    "http://freegeoip.net/json/10.0.0.10",
  ],
]
`;

exports[`index execute w/ "x-forwarded-for" 1`] = `
Array [
  Array [
    "http://freegeoip.net/json/98.6.106.250",
  ],
]
`;

exports[`index execute w/ "x-forwarded-for" w/ multiple addresses 1`] = `
Array [
  Array [
    "http://freegeoip.net/json/127.0.0.1",
  ],
]
`;
